src/index.test.ts

```diff
@@ -20,29 +20,73 @@
 // Tests for the ERC725.js package
 import assert from 'assert';
 import { hexToNumber, leftPad, numberToHex } from 'web3-utils';
-import ERC725 from '../src';
+import ERC725 from '.';
 import {
   decodeAllData,
   decodeKey,
   decodeKeyValue,
   encodeAllData,
   encodeKey,
   encodeKeyValue,
-} from '../src/lib/utils';
-import { Erc725Schema } from '../src/types/Erc725Schema';
-import { ApolloClient, EthereumProvider, HttpProvider } from './mockProviders';
-import { mockSchema } from './mockSchema';
+} from './lib/utils';
+import { Erc725Schema } from './types/Erc725Schema';
+import {
+  ApolloClient,
+  EthereumProvider,
+  HttpProvider,
+} from '../test/mockProviders';
+import { mockSchema } from '../test/mockSchema';
 import {
   generateAllData,
   generateAllRawData,
   generateAllResults,
-} from './testHelpers';
+} from '../test/testHelpers';
 
+// eslint-disable-next-line import/no-extraneous-dependencies
 import 'isomorphic-fetch';
 
 const address = '0x0c03fba782b07bcf810deb3b7f0595024a444f4e';
 
 describe('Running erc725.js tests...', () => {
+  it('should throw when no arguments are supplied', () => {
+    assert.throws(
+      () => {
+        // @ts-ignore
+        // eslint-disable-next-line no-new
+        new ERC725();
+      },
+      (error) => error.message === 'Missing schema.',
+    );
+  });
+
+  it('should throw when incorrect or unsupported provider is provided', () => {
+    assert.throws(
+      () => {
+        // @ts-ignore
+        // eslint-disable-next-line no-new
+        new ERC725(mockSchema, address, { test: false });
+      },
+      (error) =>
+        error.message.indexOf('Incorrect or unsupported provider') >= -1,
+    );
+  });
+
+  it('should throw when incorrect calling getData without address & provider options set', async () => {
```

Comment:
i don't think the word "incorrect" adds value here. Why not simply:

"should throw when calling getData without address & provider options set"